File: Code/Arrays_Bit_Manipulation/Maximum_Subarray_Kadane_S_Algorithm.py
def max_sub_array(nums):
    current_sum = max_sum = nums[0]

    for i in range(1, len(nums)):
        current_sum = max(nums[i], current_sum + nums[i])
        max_sum = max(max_sum, current_sum)

    return max_sum

if __name__ == "__main__":
    nums = [-2, 1, -3, 4, -1, 2, 1, -5, 4]
    result = max_sub_array(nums)
    print("Maximum Subarray Sum:", result)

File: Code/Arrays_Bit_Manipulation/Maximum_Subarray_Kadane_S_Algorithm.cs
using System;

class Program {
    static int MaxSubArray(int[] nums) {
        int currentSum = nums[0];
        int maxSum = nums[0];

        for (int i = 1; i < nums.Length; i++) {
            currentSum = Math.Max(nums[i], currentSum + nums[i]);
            maxSum = Math.Max(maxSum, currentSum);
        }

        return maxSum;
    }

    static void Main() {
        int[] nums = { -2, 1, -3, 4, -1, 2, 1, -5, 4 };
        int result = MaxSubArray(nums);
        Console.WriteLine("Maximum Subarray Sum: " + result);
    }
}
